hydpy/models/rconc/rconc_model.py

```diff
@@ -0,0 +1,275 @@
+        >>> model.determine_outflow_v2()
+        >>> fluxes.outflow
+        outflow(2.0)
+
+        We solve the underlying ordinary differential equation via the explicit Euler
+        method.  Nevertheless, defining arbitrarily high storage coefficients does not
+        pose any stability problems due to truncating too high outflow values:
+
+        >>> control.recstep(1)
+        >>> derived.dt.update()
+        >>> nmbstorages(5)
+        >>> derived.ksc(inf)
+        >>> model.determine_outflow_v2()
+        >>> fluxes.outflow
+        outflow(2.0)
+
+        Increasing the number of internal calculation steps via parameter |RecStep|
+        results in higher numerical accuracy without violating the water balance:
+
+        >>> derived.ksc(2.0)
+        >>> states.sc = 0.0
+        >>> model.determine_outflow_v2()
+        >>> fluxes.outflow
+        outflow(2.0)
+        >>> states.sc
+        sc(0.0, 0.0, 0.0, 0.0, 0.0)
+
+        >>> control.recstep(10)
+        >>> derived.dt.update()
+        >>> states.sc = 0.0
+        >>> model.determine_outflow_v2()
+        >>> fluxes.outflow
+        outflow(0.084262)
+        >>> states.sc
+        sc(0.714101, 0.542302, 0.353323, 0.202141, 0.103872)
+        >>> from hydpy import round_
+        >>> round_(fluxes.outflow + sum(states.sc))
+        2.0
+
+        >>> control.recstep(100)
+        >>> derived.dt.update()
+        >>> states.sc = 0.0
+        >>> model.determine_outflow_v2()
+        >>> fluxes.outflow
+        outflow(0.026159)
+        >>> states.sc
+        sc(0.850033, 0.590099, 0.327565, 0.149042, 0.057103)
+        >>> round_(fluxes.outflow + sum(states.sc))
+        2.0
+    """
+
+    CONTROLPARAMETERS = (rconc_control.NmbStorages, rconc_control.RecStep)
+    DERIVEDPARAMETERS = (rconc_derived.DT, rconc_derived.KSC)
+    REQUIREDSEQUENCES = (rconc_fluxes.Inflow,)
+    UPDATEDSEQUENCES = (rconc_states.SC,)
+    RESULTSEQUENCES = (rconc_fluxes.Outflow,)
+
+    @staticmethod
+    def __call__(model: modeltools.Model) -> None:
+        con = model.parameters.control.fastaccess
+        der = model.parameters.derived.fastaccess
+        flu = model.sequences.fluxes.fastaccess
+        sta = model.sequences.states.fastaccess
+        if (con.nmbstorages == 0) or modelutils.isinf(der.ksc):
+            flu.outflow = flu.inflow
+        else:
+            flu.outflow = 0.0
+            for _ in range(con.recstep):
+                sta.sc[0] += der.dt * flu.inflow
+                for j in range(con.nmbstorages - 1):
+                    d_q = min(der.dt * der.ksc * sta.sc[j], sta.sc[j])
+                    sta.sc[j] -= d_q
+                    sta.sc[j + 1] += d_q
+                j = con.nmbstorages - 1
+                d_q = min(der.dt * der.ksc * sta.sc[j], sta.sc[j])
+                sta.sc[j] -= d_q
+                flu.outflow += d_q
+
+
+class Set_Inflow_V1(modeltools.Method):
+    """Set the input for the calculation of the runoff concentration."""
```

Comment:
We can reuse the interface method's description: "Set the runoff concentration input in mm/T."